### DISCUSSION TITLE
Updating package-lock.json to lockfileVersion 2

### DISCUSSION THREAD

**Discussion by NicolasMassart** [tcort/link-check#41] [General]
When using `npm install` on node 15, it automatically updates the package lock file.
Should I go back to a lower node version to update so that it doesn't change the lock version or is it ok?
@tcort any opinion on this?

**Comment by nschonni** [comment]
Dependabot will roll back any upgrade on the next time it runs

**Reply by NicolasMassart** [reply]
Is it going to rollback the deps versions or just the file format from v2 to v1?

**Reply by nschonni** [reply]
I think just the lockfile changes. But that creates mutli-thousand line diffs everytime it someone tries to upgrade the version, then it gets rolled back.
Maybe they've fixed that issue though, I haven't looked at it in a month or more

**Reply by NicolasMassart** [reply]
The thing is that it's advised to use the lock file to have reproducible builds with the `npm ci` command. That's why we commit the lock file too. So if we can't modify it to match the deps updates, and I'm not going to update the lock by hand, how can we achieve this?

**Reply by NicolasMassart** [reply]
@tcort any thought about this? What prevents us to upgrade the package-lock format to `"lockfileVersion": 2`?

**Reply by nschonni** [reply]
Looks like support is there for Dependabot now https://github.blog/changelog/2021-02-08-dependabot-npm-v7-support/
Downside is that means all devs working on the project will need to upgrade to NPM 7, which can cause friction if not all their other projects have upgraded as well

**Reply by NicolasMassart** [reply]
well Node 7... we are now at 14 as an Active LTS version...
https://nodejs.org/en/about/releases/

**Reply by nschonni** [reply]
Node 14 is still shipping with NPM 6 https://nodejs.org/en/download/releases/, only Node 15 currently has NPM 7 out of the box. Although I think you might have mixed up Node and NPM in my previous comment

**Reply by NicolasMassart** [reply]
Yes I totally mixed them... Sorry.
So we keep the v1 package for now. Thanks!
